### PR TITLE
remove noisy warning in StepCounterHook

### LINKED ISSUE #21523: SyncReplicaOptimizer prints global step warning unnecessarily
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: NA
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.10.0
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: NA
- **GCC/Compiler version (if compiling from source)**: NA
- **CUDA/cuDNN version**:  cuda-9.0/ cuDNN 7
- **GPU model and memory**: TITAN X 12GB
- **Exact command to reproduce**:  Run the code below to produce the warnings

### Describe the problem
This is less a bug and more a suggested improvement. The setup illustrated in the code above is common for reinforcement learning algorithms where the workers are used to collect experience by acting in an environment according to the current policy in a distributed fashion and then synchronizing to update the parameters of the shared networks (stored on the parameter server). There are two ops that are run (multiple times) in a single iteration of the training loop. One is an `act` op and the other is the `update` op. The `act` op has nothing to do with the optimizer while the `update` op does. As such, running the `act` op should not affect the `global_step` while the `update` op should increment `global_step`. The problem, in my estimation, is that tensorflow does not recognize this difference and complains (by printing a warning) that `global_step` is not incremented every time the `act` op is run.

This is extremely annoying as thousands of warnings are printed and useful information is lost in the torrent of text. Turning off all warning is the workaround I am using right now, but this is clearly not a long-term solution.

The exact warning is:

``` 

WARNING:tensorflow:It seems that global step (tf.train.get_global_step) has not been increased. Current value (could be stable): 835 vs previous value: 835. You could increase the global step by passing tf.train.get_global_step() to Optimizer.apply_gradients or Optimizer.minimize.
```

### Source code / logs
```
import os
import multiprocessing as mp
import tensorflow as tf
import numpy as np

num_workers=4

def model(images):
    net = tf.layers.dense(images, 500, activation=tf.nn.relu)
    net = tf.layers.dense(net, 500, activation=tf.nn.relu)
    net = tf.layers.dense(net, 10, activation=None)
    return net

def train_run_parallel(cluster, job, task, num_workers):
    server = tf.train.Server(cluster, job_name=job, task_index=task)
    if job == 'ps':
        server.join()
    else:
        worker_device = "/job:worker/task:{}".format(task)
        with tf.device(tf.train.replica_device_setter(cluster=cluster)):
            def train_next_batch(batchsize):
                imgs = np.random.rand(batchsize,784)
                inds = np.random.randint(0, 10, (batchsize))
                labels = np.eye(10)[inds]
                return imgs,labels
            
            images = tf.placeholder(tf.float32, [None, 784])
            labels = tf.placeholder(tf.int32, [None, 10])    

            logits = model(images)
            loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits=logits, labels=labels))

            hooks = []
            global_step = tf.train.get_or_create_global_step()
            optimizer = tf.train.AdamOptimizer(learning_rate=1e-04)
            optimizer = tf.train.SyncReplicasOptimizer(optimizer, replicas_to_aggregate=num_workers,
                                        total_num_replicas=num_workers, )
            
            hooks.append(optimizer.make_session_run_hook(task==0, num_tokens=0))
            train_op = optimizer.minimize(loss, global_step=global_step,
                                          aggregation_method=tf.AggregationMethod.ADD_N)
            
            
            mon_sess = tf.train.MonitoredTrainingSession(master=server.target,
                                                   is_chief=(task == 0),
                                                   checkpoint_dir="./test_checkpoint_dir",
                                                   hooks=hooks)

            for e in range(1000):
                # simulate roll-out for reinforcement learning
                for _ in range(100):
                    img_batch, label_batch = train_next_batch(32)
                    pred_logits = mon_sess.run([logits], feed_dict={images:img_batch})
                img_batch, label_batch = train_next_batch(32)
                _, ls, step = mon_sess.run([train_op, loss, global_step],
                                            feed_dict={images: img_batch, labels: label_batch})
                if step % 10 == 0:
                    print("Worker %d, Train step %d, loss: %f" % (task, step, ls))
            server.join()
            
def main():    
    cluster = tf.train.ClusterSpec({
        'worker': ['localhost:'+str(30352+w) for w in range(num_workers)],
        'ps': [
            'localhost:30351'
        ]
    })

    job_task_index_map = [('ps', 0)]
    for w in range(num_workers): job_task_index_map.append(('worker', w))

    procs = []

    for job, task in job_task_index_map:
        proc = mp.Process(target=train_run_parallel, args=(cluster, job, task, num_workers))
        procs.append(proc)
        proc.start()
    
    for proc in procs:
        proc.join()

if __name__ == '__main__':
    main()
```

### PR DESCRIPTION
**Pull request by facaiy**
fix #21523

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by facaiy** [comment]
@xiejw Hi, would you mind taking a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @xiejw: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @xiejw: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 44 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @xiejw: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @xiejw: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @xiejw: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 44 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rchao: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Review by rchao — APPROVED** [review]
Thanks for the PR! This looks good to me and approved.

**Comment by facaiy** [comment]
@rchao Thanks for your review.

**Comment by facaiy** [comment]
@rthadur Thanks :-)
